src/contract.rs

```diff
@@ -1228,7 +1345,116 @@ mod tests {
     }
 
     #[test]
-    fn execute_with_invalid_data() {
+    fn execute_match_with_valid_scope_data() {
+        // setup
+        let mut deps = mock_dependencies(&[]);
+
+        let scope_input = Scope {
+            scope_id: "scope1234".to_string(),
+            specification_id: "scopespec1".to_string(),
+            owners: vec![Party {
+                address: Addr::unchecked("asker"),
+                role: PartyType::Owner,
+            }],
+            data_access: vec![],
+            value_owner_address: Addr::unchecked("asker"), // todo: does this need to be the contract's address?
```

Comment:
Once my change goes through, yessir.  Use `MOCK_CONTRACT_ADDR` for that, which is the default thingy in cosmwasm, and used by `mock_env()`